modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/table/ClientSchemasGetRequest.java

```diff
@@ -43,29 +45,38 @@ public class ClientSchemasGetRequest {
     public static CompletableFuture<Void> process(
             ClientMessageUnpacker in,
             ClientMessagePacker out,
-            IgniteTables tables
+            IgniteTables tables,
+            SchemaVersions schemaVersions
     ) {
-        return readTableAsync(in, tables).thenAccept(table -> {
+        return readTableAsync(in, tables).thenCompose(table -> {
             if (in.tryUnpackNil()) {
                 // Return the latest schema.
                 out.packInt(1);
 
-                var schema = table.schemaView().schema();
+                return schemaVersions.schemaVersionAtNow(table.tableId())
+                        .thenAccept(version -> {
+                            SchemaDescriptor schema = table.schemaView().schemaNow(version);
```

Comment:
`schemaNow()` is named analogously to `CompletableFuture#getNow()`: it returns whatever it has 'right now', without any additional waits/blocks. But `now` in `schemaVersionAtNow()` has a different meaning: it's a schema version that corresponds to T1=now() in terms of Schema Sync: that is, we might wait for some time to make sure that all schemas needed for T1 are available on the current node.

So `schemaVersionAtNow()` makes sure we get a correct (from the point of view of Schema Sync) schema version number, and then `schemaNow()` is used to obtain a schema by this number.

There is a problem with naming (two meanings for same word 'now'), probably this can be improved. I'll try to invent something.

Reply:
It's clearer now, thank you!
We personally discussed renaming.

Reply:
`schemaNow(int)` was renamed back to `schema(int)`, so the ambiguity should have been gone now.